File: cs/GenericsComparison/Complex/CustomNode.cs
using System;

namespace Complex
{
    public class CustomNode : Node<int>
    {
        public CustomNode(int data) : base(data)
        {
        }
    }
    
    
    public class Program
    {
        public static void Main(string[] args)
        {
            var node = new CustomNode(10);
        }
    }
}

File: java/src/compiled/erasure/CustomNode.java
package compiled.erasure;

public class CustomNode extends Node {
    public CustomNode(Integer data) { super(data); }

    public void setData(Object data) {
        setData((Integer) data);
    }

    public void setData(Integer data) {
        System.out.println("MyNode.setData");
        super.setData(data);
    }
}